dbms/src/Flash/Mpp/ReceiverChannelWriter.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Flash/Mpp/ReceiverChannelWriter.h>
+
+namespace DB
+{
+namespace
+{
+inline void injectFailPointReceiverPushFail(bool & push_succeed, ExchangeMode mode)
+{
+    switch (mode)
+    {
+    case ExchangeMode::Local:
+        fiu_do_on(FailPoints::random_receiver_local_msg_push_failure_failpoint, push_succeed = false);
```

Comment:
Would we add `FailPoints::random_receiver_local_msg_push_failure_failpoint` in random failpoint test case later?

Reply:
> Would we add `FailPoints::random_receiver_local_msg_push_failure_failpoint` in random failpoint test case later?

okk